specification/eventgrid/typespec/main.tsp

```diff
@@ -0,0 +1,230 @@
+  model LockToken {
+    @doc("The token used to lock the event.")
+    lockToken: string;
+  }
+
+  @doc("Properties of the Event Broker operation.")
+  model BrokerProperties {
+    @doc("The token used to lock the event.")
+    lockToken: LockToken;
+  }
+
+  @doc("Receive operation details per Cloud Event.") 
+  model ReceiveDetails {
+    @doc("The Event Broker details.")
+    brokerProperties: BrokerProperties;
+
+    @doc("Cloud Event details.")
+    event: CloudEventEvent;
+  }
+
+  @doc("Details of the Receive operation response.")
+  model ReceiveResponse {
+    @doc("Array of receive responses, one per cloud event.")
+    value: ReceiveDetails[];
+  }
+   
+  @doc("Failed LockToken information.")
+  model FailedLockToken {
+    @doc("LockToken value")
+    lockToken: LockToken;
+
+    @doc("Error code")
+    errorCode: int32;
+
+    @doc("Description of the error")
+    errorDescription: string; 
+  }
+   
+  @doc("Details of the LockTokens information. This is used for both Acknowledge and Release operation response.")
+  model LockTokensResponse {
+    @doc("Array of LockToken values for failed cloud events.")
+    failedLockTokens: FailedLockToken[];
+
+    @doc("Array of LockToken values for succeeded cloud events.")
+    succeededLockTokens: string[];
+  }
+
+  // POST https://{namespaceName}.{region}.eventgrid.azure.net/topics/{topicName}:publish?api-version={apiVersion}}
+
+  @doc("Publish Single Cloud Event to namespace topic.")
+  @route("/topics/{topicName}:publish", {shared: true})
+  @post op PublishCloudEvent is Azure.Core.RpcOperation<{
+    @doc("content type")
+    @header("content-type")
+    contentType: "application/cloudevents+json; charset=utf-8";
+
+    @doc("Topic Name.")
+    @path
+    topicName: string;
+
+    @doc("Single Cloud Event being published.")
+    @body 
+    event: CloudEventEvent;
+  }, OkResponse>;
+
+
+  @doc("Publish Batch of Cloud Events to namespace topic.")
+  @route("/topics/{topicName}:publish", {shared: true})
+  @post op PublishBatchOfCloudEvents is Azure.Core.RpcOperation<{
+    @doc("content type")
+    @header("content-type")
+    contentType: "application/cloudevents-batch+json; charset=utf-8";
+
+    @doc("Topic Name.")
+    @path
+    topicName: string;
+
+    @doc("Array of Cloud Events being published.")
+    @body
+    events: CloudEventEvent[];
+  }, OkResponse>;
+   
+  @doc("Receive Batch of Cloud Events from the Event Subscription.")
+  @route("/topics/{topicName}/eventsubscriptions/{eventSubscriptionName}:receive")
+  @post op ReceiveBatchOfCloudEvents is Azure.Core.RpcOperation<{
+    @doc("Topic Name.")
+    @path
+    topicName: string;
+
+    @doc("Event Subscription Name.")
+    @path
+    eventSubscriptionName: string;
+
+    @doc("Max Events count to be received.")
+    @query
+    maxEvents: int32;
+
+    @doc("Timeout value for receive operation in Seconds. Default is 60 seconds.")
+    @query
+    timeout: int32;
+  }, ReceiveResponse & CreatedResponse >;
```

Comment:
What we see calling the service, is that nothing to receive returns an empty list with a 200, and if there are events this is a 201